NineChronicles.Headless.Tests/GraphTypes/States/Models/ShopStateTypeTest.cs

```diff
@@ -0,0 +1,180 @@
+                    }
+                }
+            }";
+
+            var action = new SetAvatarState();
+            BlockChain.MakeTransaction(
+                userPrivateKey,
+                new PolymorphicAction<ActionBase>[] { action }
+            );
+            await BlockChain.MineBlock(minerAddress);
+
+            var avatarState = new AvatarState((Dictionary)BlockChain.GetState(avatarAddress));
+            var expected = new Dictionary<string, object>
+            {
+                ["stateQuery"] = new Dictionary<string, object>
+                {
+                    ["shop"] = new Dictionary<string, object>
+                    {
+                        ["address"] = Addresses.Shop.ToString(),
+                        ["products"] = new List<object>
+                        {
+                            new Dictionary<string, object>
+                            {
+                                ["sellerAgentAddress"] = "0xfc2a412ea59122B114B672a5518Bc113955Dd2FE",
+                                ["sellerAvatarAddress"] = "0x983c3Fbfe8243a0e36D55C6C1aE26A7c8Bb6CBd4",
+                                ["price"] = "100 NCG",
+                                ["itemUsable"] = itemUsable,
+                                ["costume"] = costume,
+                            }
+                        },
+                    },
+                },
+            };
+
+            var action2 = new Sell3
+            {
+                sellerAvatarAddress = avatarAddress,
+                itemId = itemId,
+                price = new Currency("NCG", 2, minter: null) * 100
+            };
+            BlockChain.MakeTransaction(
+                userPrivateKey,
+                new PolymorphicAction<ActionBase>[] { action2 }
+            );
+            await BlockChain.MineBlock(minerAddress);
+            var queryResult = await ExecuteQueryAsync(query);
+            Assert.Equal(expected, queryResult.Data);
+        }
+
+        public static IEnumerable<object[]> Members => new List<object[]>
+        {
+            new object[]
+            {
+                new Guid("220acb43-095e-46f6-9725-4223c69827e8"),
+                new Dictionary<string, object>
+                {
+                    ["itemId"] = new Guid("220acb43-095e-46f6-9725-4223c69827e8"),
+                    ["itemType"] = ItemType.Equipment.ToString().ToUpper(),
+                    ["itemSubType"] = ItemSubType.Weapon.ToString().ToUpper(),
+                },
+                null,
+            },
+            new object[]
+            {
+                new Guid("d3d9ac06-eb91-4cc4-863a-5b4769ad633e"),
+                null,
+                new Dictionary<string, object>
+                {
+                    ["itemId"] = new Guid("d3d9ac06-eb91-4cc4-863a-5b4769ad633e"),
+                    ["itemType"] = ItemType.Costume.ToString().ToUpper(),
+                    ["itemSubType"] = "FULL_COSTUME",
+                },
+            },
+        };
+    }
+}
```

Comment:
IMO I believe these tests can have more small scope without `action` and `blockchain` and their names can be more expressionable without `StateQuery` prefix.

I want to suggest to test with [`GraphQLTestUtils`](https://github.com/planetarium/NineChronicles.Headless/blob/738d21d23c3a895e40f3bbd6ef57fd0f44e125e2/NineChronicles.Headless.Tests/GraphQLTestUtils.cs#L10-L28), not `GraphQLTestBase`. The static class helps to execute query with a `IObjectGraphType` as root query easily. Then it can be free from `BlockChain<T>` and be away from not related graph types in implementation (e.g. `StateQuery`), also the `StateQuery` prefix on all tests can be removed. 😀 

Of course, I think this opinion can be wrong anytime because I didn't understand actions and states of `Lib9c` well yet. You can leave a comment in free. And you can also with call (e.g. Slack) if you want to ask or discuss this.